[REFACTOR] #32 - 리팩토링 진행(AddWord)

LEEBUYONG opened the pull request:
## 📌 관련 이슈
<!-- 관련있는 이슈 번호(#000)을 적어주세요.
  해당 pull request merge와 함께 이슈를 닫으려면 
  closed: #Issue_number를 적어주세요 -->
closed: #32 
  

## 📌 변경 사항 및 이유
<!-- 변경한 내용과 그 이유를 적어주세요. -->
ModalVC CallBack 코드 정리, AddWordModalVC 코드 간소화, DIContainer 수정

## 📌 구현 내역 스크린샷
<!-- 작업한 화면이 있다면 스크린 샷으로 첨부해주세요. -->
이전과 똑같이 동작하므로 스크린샷 생략


## 📌 PR Point
<!-- 리뷰어 분들이 집중적으로 보셨으면 하는 내용을 적어주세요 -->
UseCase 3개로 분리 후 VM,VC 로직 간소화 및 리팩토링 진행했습니다.
CoreData에 Source 삭제 후 UnsolvedWordRepositoryImpl에서 toEntity를 source: .database로 수정했습니다.

LEEBUYONG committed:
refactor: DI 추가, ModalVC 정리

- CallBack 코드 정리, AddWordModalVC 코드 간소화

gnoes-ios commented on WordPalette/WordPalette/App/SceneDelegate.swift:
```diff
@@ -14,7 +14,7 @@ class SceneDelegate: UIResponder, UIWindowSceneDelegate {
         guard let windowScene = (scene as? UIWindowScene) else { return }
         window = UIWindow(windowScene: windowScene)
-        window?.rootViewController = AddWordViewController(selectedLevel: .beginner)
+        window?.rootViewController = HomeViewController()
```
makeTabBarController로 수정해야될거 같아요

LEEBUYONG replied:
넵 수정해뒀습니다!

gnoes-ios reviewed (approved):
고생하셨습니다. 충돌나는 부분만 수정하시면 될거 같아요

LEEBUYONG committed:
fix: rootVC 변경

Sn8Ch0 commented on WordPalette/WordPalette/Presentation/AddWord/ViewController/AddWordModalViewController.swift:
```diff
@@ -49,107 +44,74 @@ final class AddWordModalViewController: UIViewController {
             DispatchQueue.main.async {
                 if isDuplicate {
-                    self?.showAlert(message: self?.duplicateMessage(for: level) ?? "이미 등록된 단어입니다.")
+                    let message = level != nil ? "\(level!.rawValue)에 이미 있는 단어입니다." : "이미 등록된 단어입니다."
```
개인적으로 (level != nli) 이렇게 표현하면 코드 가독성이 더 좋을 것 같습니다!

LEEBUYONG replied:
수정했습니다!

Sn8Ch0 reviewed (approved):
고생하셨습니다!

LEEBUYONG committed:
fix: pr 내용 반영

let message = (level != nil) ? "\(level!.rawValue)에 이미 있는 단어입니다." : "이미 등록된 단어입니다."
로 수정

LEEBUYONG committed:
Merge remote-tracking branch 'origin/dev' into refactor/#32-AddWordView
